Meme/Meme/MemeTemplatesViewModel.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  MemeTemplatesViewModel.swift
+//  Meme
+//
+//  Created by Mesfin Bekele Mekonnen on 7/30/19.
+//  Copyright © 2019 Mesfin Bekele Mekonnen. All rights reserved.
+//
+
+import Foundation
+
+protocol TemplatesViewModel {
+    func requestTemplates(completion: @escaping ([Template]) -> Void?)
+    func getMeme(template: Template) -> Meme?
+}
+
+struct MemeTemplatesViewModel: TemplatesViewModel {
```

Comment:
@vhart Thoughts on defining this as a struct instead of a class? It was a class previously. I've read arguments for both online, but I also want to get your opinion on how you go about defining your VMs?

Reply:
I usually use a `class`. I defer to `struct` if it's just a read-only model - that is, if it performs no actions or consequence or mutation. If it mutates or performs async tasks then I just use a class.